frontend/catalyst/pennylane_extensions.py

```diff
@@ -869,9 +869,19 @@ def check_is_iterable(x, hint):
         fn = _ensure_differentiable(f)
         grad_params = _check_grad_params(method, scalar_out, h, argnum)
 
-        jaxpr, _ = _make_jaxpr_check_differentiable(fn, grad_params, *params)
+        jaxpr, out_tree = _make_jaxpr_check_differentiable(fn, grad_params, *params)
 
         results = jvp_p.bind(*params, *tangents, jaxpr=jaxpr, fn=fn, grad_params=grad_params)
+
+        if out_tree.children() != []:
+            midpoint = len(results) // 2
+            func_res = results[:midpoint]
+            jvps = results[midpoint:]
+            func_res = tree_unflatten(out_tree, func_res)
+            jvps = tree_unflatten(out_tree, jvps)
+            results = tuple([func_res, jvps])
+        else:
+            results = tuple(results)
```

Comment:
Is this not inconsistent? I.e. in one branch the results and jvps are flat, but in the other they are separated into a pair?

Reply:
Yes that was the behaviour without any tree that I have kept, I will double check that.

Reply:
Yep can confirm, the motivation was to match JAX (even if there were breaking changes!)

Reply:
Yes let me double check that, because I have rework all the test to make sure the pytrees are the same and they are.

Reply:
done in #508